src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -146,6 +175,80 @@ bool GTUtilsDashboard::hasNotifications(HI::GUITestOpStatus &os) {
     return notificationsWidget->isVisible();
 }
 
+#define GT_METHOD_NAME "getNotificationTypeFromHtml"
+GTUtilsDashboard::Notification::Type GTUtilsDashboard::getNotificationTypeFromHtml(HI::GUITestOpStatus &os,
+    const QString &html) {
+    using Notification = U2::GTUtilsDashboard::Notification;
+    const QString errMsg = "Dashboard notification type not found";
+    GT_CHECK_RESULT(!html.isEmpty(), errMsg, Notification::Type::ERR)
+
+    QXmlStreamReader reader(html);
+    while (reader.readNextStartElement()) {
+        if (reader.name() == "img") {
+            const QVariant type = Notification::stringToType(reader.attributes().value("class").toString());
+            if (type.canConvert<int>()) {
+                return static_cast<Notification::Type>(type.toInt());
+            }
+        }
+    }
+    GT_CHECK_RESULT(false, errMsg, Notification::Type::ERR)
+}
+#undef GT_METHOD_NAME
+
+#define GT_METHOD_NAME "getNotificationCellText"
+QString GTUtilsDashboard::getNotificationCellText(HI::GUITestOpStatus &os, const QGridLayout &tableLayout,
+    const int row, const int col) {
+    const QWidget *cellWidget = tableLayout.itemAtPosition(row, col)->widget();
+    QString text;
+
+    if (cellWidget != nullptr && cellWidget->objectName() == "tableCell") {
+        if (const QLayout *const cellLayout = cellWidget->layout()) {
+            for (int labelInd = 0; labelInd < cellLayout->count(); ++labelInd) {
+                if (const auto *const label = qobject_cast<QLabel *>(cellLayout->itemAt(labelInd)->widget())) {
+                    text = label->text();
+                }
+            }
+        }
+    }
+    GT_CHECK_RESULT(!text.isEmpty(),
+        QString("Error getting (%1,%2) cell of dashboard notification table").arg(row).arg(col), text)
+    return text;
+}
+#undef GT_METHOD_NAME
+
+#define GT_METHOD_NAME "getNotifications"
+QList<GTUtilsDashboard::Notification> GTUtilsDashboard::getNotifications(GUITestOpStatus &os) {
+    const QString notificationsWidgetName = "NotificationsDashboardWidget";
+    const auto tableLayout = qobject_cast<QGridLayout *>(
+        GTWidget::findWidget(os, notificationsWidgetName, GTUtilsDashboard::getDashboard(os))->layout());
+    QList<Notification> notifications;
+
+    GT_CHECK_RESULT(tableLayout != nullptr && tableLayout->columnCount() == 3,
+                    notificationsWidgetName % " was found, but cannot be used in a test",
+                    notifications)
+
+    const int notificationsNumber = tableLayout->rowCount();
+    notifications.reserve(notificationsNumber);
```

Comment:
please remove optimization with reserve() unless you have a proof that this optimization is needed